function-server/src/main/java/io/dispatchframework/javabaseimage/SimpleFunctionExecutor.java

```diff
@@ -67,14 +75,40 @@ public String execute(String message) {
             }
 
             if (err == null) {
+                Map<Object, Object> context = (Map<Object, Object>) req.getContext();
+                Object payload = req.getPayload();
+                Double timeout = (Double) context.getOrDefault("timeout", 0);
+                Callable<Object> callable = () -> {
+                    return f.apply(context, payload);
+                };
+
+                ExecutorService executorService = Executors.newSingleThreadExecutor();
+                Future<Object> future = executorService.submit(callable);
```

Comment:
I like this idea. I'd wrap it into a separate method though. I'd have that method receive function, context and payload as arguments and return the same result as `f.apply(context, payload)`. 

So, in the `try {}` block (see deleted code :)), instead of 
```java
r = f.apply(req.getContext(), req.getPayload());
```
we'd have something like:
```java
r = withTimeout(f, req.getContext(), req.getPayload());
```

This new method would additionally throw `java.util.concurrent.TimeoutException` which we'd catch just before `catch (Exception ex)`.

This additional method can (and should) be unit-tested.

Let's put some timeout (the default one - if nothing is specified) into the context on the function-manager side, so we always have a timeout in the context at this point. And let's treat unspecified timeout as a SystemError. 

An indefinite timeout (`timeout == 0`) is useful, but it should be explicit.